tests/unit/ui/components/TestConfirmButton.js

```diff
@@ -107,41 +117,59 @@ describe(__filename, () => {
   });
 
   it('closes the confirmation panel on cancel ', () => {
+    const { store } = dispatchClientMetadata();
     const onConfirmSpy = sinon.spy();
-    const root = render({ onConfirm: onConfirmSpy });
+    const root = render({ onConfirm: onConfirmSpy, store });
 
     expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
 
     // Show the confirmation panel.
     root.find(Button).simulate('click', createFakeEvent());
+    applyUIStateChanges({ root, store });
 
     expect(root).toHaveClassName('ConfirmButton--show-confirmation');
     sinon.assert.notCalled(onConfirmSpy);
 
     root
       .find('.ConfirmButton-cancel-button')
       .simulate('click', createFakeEvent());
+    applyUIStateChanges({ root, store });
 
     expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
     sinon.assert.notCalled(onConfirmSpy);
   });
 
   it('calls the onConfirm prop when user clicks the "confirm" button', () => {
+    const { store } = dispatchClientMetadata();
     const onConfirmSpy = sinon.spy();
-    const root = render({ onConfirm: onConfirmSpy });
+    const root = render({ onConfirm: onConfirmSpy, store });
 
     expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
 
     // Show the confirmation panel.
     root.find(Button).simulate('click', createFakeEvent());
+    applyUIStateChanges({ root, store });
 
     expect(root).toHaveClassName('ConfirmButton--show-confirmation');
     sinon.assert.notCalled(onConfirmSpy);
 
     const event = createFakeEvent();
     root.find('.ConfirmButton-confirm-button').simulate('click', event);
+    applyUIStateChanges({ root, store });
 
     expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
     sinon.assert.calledWith(onConfirmSpy, event);
   });
+
+  it('changes UI state when the confirm button is clicked', () => {
+    const { store } = dispatchClientMetadata();
+    const root = render({ store });
+
+    expect(root.instance().props.uiState.showConfirmation).toEqual(false);
+
+    root.find(Button).simulate('click', createFakeEvent());
+    applyUIStateChanges({ root, store });
+
+    expect(root.instance().props.uiState.showConfirmation).toEqual(true);
```

Comment:
Not sure you really this test case, because it should be already tested by the previous test cases.

Reply:
ok I was wondering but I saw an explicit test for this in the other place it was already set up. But I can def removed if it's overkill here..

Reply:
> vut I saw an explicit test for this in the other place it was already set up

ah, really? I looked too, and found nothing 😅

Reply:
sorry I should have posted where I saw this (I was being lazy :/), but [here](https://github.com/mozilla/addons-frontend/blob/master/tests/unit/amo/components/TestEditableCollectionAddon.js#L224)

Reply:
mmh interesting. I'd argue that we should not be testing this because we should focus on the behaviors. If the value of this prop changes, then we should observe "something", in this case a text with two buttons should be either displayed or hidden.

But maybe @kumar303 has a different opinion.